### PR TITLE
Fix Jenkins update script.

### PR DESCRIPTION
**Pull request by KakersUK**
Seems like the Jenkins page content has changed causing the script to fail. Might be better to use the jenkins mirrors to also populate the $version.

### CODE REVIEW

File: automatic/jenkins/update.ps1 [hunk 1]
```diff
@@ -15,16 +15,13 @@ function global:au_SearchReplace {
...
 
 function global:au_GetLatest {
-    # needed for Invoke-WebRequest to work for this site
-    [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
-
     $page = Invoke-WebRequest -Uri $releases -UseBasicParsing
-    $regexUrl = "Deploy Jenkins`n(?<version>[\d\.]+)"
-    $page.RawContent -match $regexUrl
-    $version = $matches.version
+    $regex  = 'jenkins-.+\.zip$'
```
Comment (pauby) [inline automatic/jenkins/update.ps1:19]: Can you change the code to match the format that is above. There are a myriad of ways to do this but I prefer the way I have it as it's much simpler to read.

**Comment by pauby** [comment]
@KakersUK Thank you for picking this up. I dropped the ball on it!

**Comment by KakersUK** [comment]
Would this be more acceptable? Appoligies I'm fairly new to Powershell and Chocolatey AU. Hopefully the revised code is more to your liking.

Just for some context I went for [mirrors](http://mirrors.jenkins-ci.org/windows-stable/) page instead of the main Jenkins [download](https://jenkins.io/download/) page as the downloads page currently shows a version not yes available from the mirrors, which of course will cause the build to fail.

I also needed to add some filtering to the code as the mirrors page has not only many different versions but also uses the .zip and .zip.sha256 extensions.
`$pageFiltered = $page.links | Where-Object href -match '\.zip$' | Select-Object -Last 1 -expand href`

**Comment by pauby** [comment]
Apologies for taking so long to come back to this!

Jenkins have changed the way they are storing / downloading the binaries again. From version 2.235.3 they are using folders with a Jenkins.msi inside.

I'm going to close this PR as it's no longer reflecting what they do. I'm also going to embed the MSI inside the package.

Thanks for all the work you put into this. It is really appreciated.

**Comment by pauby** [comment]
Just as an FYI, I've submitted the new version of Jenkins (2.235.5) just now.

**Pull request closed without merging**